rust/hyperlane-base/src/contract_sync/mailbox.rs

```diff
@@ -98,107 +95,119 @@ where
         // Note this means we only handle this case upon observing messages in some
         // range [C,D] that indicate a previously indexed range may have
         // missed some messages.
-        tokio::spawn(async move {
-            let mut cursor = cursor.await?;
-
+        let mut cursor = cursor.await?;
+        let start_block = cursor.current_position();
+        let mut last_valid_range_start_block = start_block;
+        info!(
+            from = start_block,
+            "Resuming indexer from latest valid message range start block"
+        );
+        indexed_height.set(start_block as i64);
+
+        loop {
             let start_block = cursor.current_position();
-            let mut last_valid_range_start_block = start_block;
-            info!(from = start_block, "Resuming indexer from latest valid message range start block");
-            indexed_height.set(start_block as i64);
-
-            loop {
-                let start_block = cursor.current_position();
-                let (from, to) = match cursor.next_range().await {
-                    Ok(range) => range,
-                    Err(err) => {
-                        warn!(error = %err, "Failed to get next block range");
-                        continue;
-                    }
-                };
-
-                let mut sorted_messages: Vec<_> = indexer
-                    .fetch_sorted_messages(from, to)
-                    .await?
-                    .into_iter()
-                    .map(|(msg, _)| msg)
-                    .collect();
-
-                debug!(from, to, message_count = sorted_messages.len(), "Indexed block range");
-
-                // Get the latest known nonce. All messages whose indices are <= this index
-                // have been stored in the DB.
-                let last_nonce = db.retrieve_latest_nonce()?;
-
-                // Filter out any messages that have already been successfully indexed and stored.
-                // This is necessary if we're re-indexing blocks in hope of finding missing messages.
-                if let Some(min_nonce) = last_nonce {
-                    sorted_messages.retain(|m| m.nonce > min_nonce);
-                }
+            let Ok((from, to, eta)) = cursor.next_range().await else { continue };
+
+            let mut sorted_messages: Vec<_> = self
+                .indexer
+                .fetch_sorted_messages(from, to)
+                .await?
+                .into_iter()
+                .map(|(msg, _)| msg)
+                .collect();
+
+            info!(
+                from,
+                to,
+                distance_from_tip = cursor.distance_from_tip(),
+                estimated_time_to_sync = fmt_duration(eta),
```

Comment:
a little worried that the presence of this in the logs could lead ppl to believe they're not fully synced, even if the time is saying 0 seconds. Blockchain nodes for instance usually only log this when they're behind, so I could see some ppl bringing that knowledge into here and incorrectly believing there's something wrong and their relayer is unable to fully sync or something.

Just calling out in case you think there may be an easy / nice way to avoid the chance of this confusion, otherwise I'm happy to merge with this log and we can always change if ppl seem confused

Reply:
Changed it to say "synced" for the ETA when it is `0`